spec/models/contract_period_spec.rb

```diff
@@ -79,4 +79,22 @@
       end
     end
   end
+
+  describe '.most_recent_first' do
+    let!(:period_2022) { FactoryBot.create(:contract_period, year: 2022, started_on: Date.new(2022, 6, 1), finished_on: Date.new(2023, 5, 31)) }
+    let!(:period_2024) { FactoryBot.create(:contract_period, year: 2024, started_on: Date.new(2024, 6, 1), finished_on: Date.new(2025, 5, 31)) }
+    let!(:period_2023) { FactoryBot.create(:contract_period, year: 2023, started_on: Date.new(2023, 6, 1), finished_on: Date.new(2024, 5, 31)) }
+    let!(:period_2025) { FactoryBot.create(:contract_period, year: 2025, started_on: Date.new(2025, 6, 1), finished_on: Date.new(2026, 5, 31)) }
+
+    it 'orders contract periods by year in descending order' do
+      result = ContractPeriod.most_recent_first
+      expect(result.to_a).to eq([period_2025, period_2024, period_2023, period_2022])
+    end
+
+    it 'returns contract periods with most recent year first' do
+      result = ContractPeriod.most_recent_first
+      expect(result.first).to eq(period_2025)
+      expect(result.last).to eq(period_2022)
+    end
```

Comment:
Might be worth adding a test for `started_on: :desc` too just to make sure it's not accidentally removed in future.